BlueprintUI/Sources/Blueprint View/ElementPreview.swift

```diff
@@ -0,0 +1,240 @@
+///
+/// Why? Xcode uses a new feature called "Dynamic Replacement" to re-compile the source file you are editing,
+/// and inject it back into the running app which drives the preview. This only works on the level of a single
+/// file – if your preview and element live in separate files, Xcode needs to recompile your entire module
+/// which will slow down preview updates greatly.
+///
+/// You can learn more about Xcode previews here: https://nshipster.com/swiftui-previews/
+///
+/// Requirements
+/// --------
+/// You must be running Xcode 11 and Catalina to take advantage of live previews.
+/// They do not work on Mojave. Your selected simulator must also be an iOS 13 device.
+///
+@available(iOS 13.0, *)
+public struct ElementPreview : View {
+    
+    // MARK: Properties
+    
+    /// A provider which returns a new element.
+    public typealias ElementProvider = () -> Element
+    
+    /// The types of previews to include in the Xcode preview.
+    private var previewTypes : [PreviewType]
+    
+    /// The provider which vends a new element.
+    private var provider : ElementProvider
+    
+    // MARK: Initialization
+    
+    
+    /// Creates a new `ElementPreview` with several common devices that your users may use.
+    public static func commonDevices(with provider : @escaping ElementProvider) -> Self {
+        return Self(
+            with: [
+                .device(.iPhoneSE),
+                .device(.iPhone8),
+                .device(.iPhone8Plus),
+                .device(.iPhoneXs),
+                .device(.iPhoneXsMax),
+                
+                .device(.iPadPro_9_7),
+                .device(.iPadPro_10_5),
+                .device(.iPadPro_12_9_3),
+            ],
+            with: provider
+        )
+    }
+    
+
+    /// Creates a new `ElementPreview` with the provided preview type.
+    /// If you do not pass a preview type, `.thatFits` is used.
+    public init(
+        with previewType : PreviewType = .thatFits,
+        with provider : @escaping ElementProvider
+    ) {
+        self.init(
+            with: [previewType],
+            with: provider
+        )
+    }
+    
+    /// Creates a new `ElementPreview` with the provided preview types.
+    ///
+    /// You can pass as many preview types as you would like to see your
+    /// element rendered in those different environments.
+    ///
+    /// If you do not pass a preview type, `.thatFits` is used.
+    public init(
+        with previewTypes : [PreviewType] = [.thatFits],
+        with provider : @escaping ElementProvider
+    ) {
+        self.previewTypes = previewTypes
+        self.provider = provider
+    }
+    
+    // MARK: View
+    
+    public var body: some View {
+        ForEach(self.previewTypes, id: \.self) { previewType in
+            previewType.preview(for: ElementView(element: self.provider()))
+        }
+    }
+    
+    private struct ElementView : UIViewRepresentable {
```

Comment:
Not super related to this PR, but good example of how we can mix and match Blueprint into SwiftUI.